x-pack/plugins/lens/server/routes/field_stats.ts

```diff
@@ -75,11 +82,13 @@ export async function initFieldsRoute(setup: CoreSetup<PluginStartContract>) {
 
         const search = async (aggs: unknown) => {
           const { body: result } = await requestClient.search({
-            index: req.params.indexPatternTitle,
+            index: indexPattern.title,
             track_total_hits: true,
             body: {
               query,
               aggs,
+              runtime_mappings:
+                !field.isMapped && field.runtimeField ? { [fieldName]: field.runtimeField } : {},
```

Comment:
I am not 100% sure if this is the "correct" behavior for runtime fields shadowing mapped fields. @mattkime Will runtime fields shadowing mapped fields report as `isMapped: true` and `runtimeField: true` (that was my understanding so far). If it's the case we should not check on `isMapped` here and always write the script if `runtimeField === true`.

Reply:
Good point Tim, fixed this.

Reply:
Tim's understanding is correct.